nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,324 @@
+        .name("azure-cosmos-db-key")
+        .displayName("Azrue Cosmos DB Access Key")
+        .description("Azure Cosmos DB Access Key from Azure Portal (Settings->Keys)")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .sensitive(true)
+        .build();
+
+    static final PropertyDescriptor DATABASE_NAME = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-db-name")
+        .displayName("Azure Cosmos DB Name")
+        .description("The name of the database to use")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CONTAINER_ID = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-container-id")
+        .description("Azure Cosmos DB Container Id to use")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CONSISTENCY = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-consistency-level")
+        .displayName("Azure Cosmos DB Consistency Level")
+        .description("Azure Cosmos DB Consistency Level to use")
+        .required(false)
+        .defaultValue(CONSISTENCY_SESSION)
+        .allowableValues(CONSISTENCY_STRONG, CONSISTENCY_BOUNDED_STALENESS, CONSISTENCY_SESSION,
+                CONSISTENCY_CONSISTENT_PREFIX, CONSISTENCY_EVENTUAL)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor PARTITION_KEY = new PropertyDescriptor.Builder()
+        .name("azure-cosmos-partition-key")
+        .description("Partition Key Field Name defined during Azure Cosmos Container Creation Time")
+        .required(true)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .build();
+
+    static final PropertyDescriptor CHARACTER_SET = new PropertyDescriptor.Builder()
+        .name("charactor-set")
+        .description("The Character Set in which the data is encoded")
+        .required(false)
+        .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+        .defaultValue("UTF-8")
+        .build();
+
+    static final List<PropertyDescriptor> descriptors;
+
+    static {
+        List<PropertyDescriptor> _temp = new ArrayList<>();
+        _temp.add(CONNECTION_SERVICE);
+        _temp.add(URI);
+        _temp.add(DB_ACCESS_KEY);
+        _temp.add(CONSISTENCY);
+        _temp.add(DATABASE_NAME);
+        _temp.add(CONTAINER_ID);
+        _temp.add(PARTITION_KEY);
+        descriptors = Collections.unmodifiableList(_temp);
+    }
+
+    protected CosmosClient cosmosClient;
+    protected CosmosContainer container;
+    protected AzureCosmosDBConnectionService connectionService = null;
+
+    @OnScheduled
+    public void createClient(ProcessContext context) throws Exception {
+        final String uri, accessKey;
```

Comment:
These don't get used out of the `else` so I'd just move them inside.

Reply:
moved inside else clause